ScaleformUI_Csharp/UIMenu.cs

```diff
@@ -1969,28 +1976,78 @@ public async void ProcessControl(Keys key = Keys.None)
             if (MenuItems.Count == 0) return;
             if (IsControlBeingPressed(MenuControls.Up, key) && UpdateOnscreenKeyboard() != 0 && !IsWarningMessageActive())
             {
-                GoUp();
+                if (GetGameTimer() - time > delay)
+                {
+                    ButtonDelay();
+                    GoUp();
+                }
             }
 
             else if (IsControlBeingPressed(MenuControls.Down, key) && UpdateOnscreenKeyboard() != 0 && !IsWarningMessageActive())
             {
-                GoDown();
+                if (GetGameTimer() - time > delay)
+                {
+                    ButtonDelay();
+                    GoDown();
+                }
             }
 
             else if (IsControlBeingPressed(MenuControls.Left, key) && UpdateOnscreenKeyboard() != 0 && !IsWarningMessageActive())
             {
-                GoLeft();
+                if (GetGameTimer() - time > delay)
+                {
+                    ButtonDelay();
+                    GoLeft();
+                }
             }
 
             else if (IsControlBeingPressed(MenuControls.Right, key) && UpdateOnscreenKeyboard() != 0 && !IsWarningMessageActive())
             {
-                GoRight();
+                if (GetGameTimer() - time > delay)
+                {
+                    ButtonDelay();
+                    GoRight();
+                }
             }
 
             else if (HasControlJustBeenPressed(MenuControls.Select, key) && UpdateOnscreenKeyboard() != 0 && !IsWarningMessageActive())
             {
                 Select(true);
             }
+
+            // IsControlBeingPressed doesn't run every frame so I had to use this
+            if (!Game.IsControlPressed(0, Control.FrontendUp) && !Game.IsControlPressed(0, Control.FrontendDown) && !Game.IsControlPressed(0, Control.FrontendLeft) && !Game.IsControlPressed(0, Control.FrontendRight))
```

Comment:
also.. for reference.. you could have used HasControlJustBeenReleased in the ScaleformUI API itself for this.
Check here https://github.com/manups4e/ScaleformUI/blob/master/ScaleformUI_Csharp/UIMenu.cs#L2000